src/app/shared/filter/date/prev-date-range.component.ts

```diff
@@ -0,0 +1,112 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Subscription } from 'rxjs/Subscription';
+import * as moment from 'moment';
+import { FilterModel, TODAY, THIS_WEEK, TWO_WEEKS, THIS_MONTH, THREE_MONTHS, THIS_YEAR,
+  YESTERDAY, LAST_WEEK, PRIOR_TWO_WEEKS, LAST_MONTH, PRIOR_THREE_MONTHS, LAST_YEAR } from '../../../ngrx/model';
+import { selectFilter } from '../../../ngrx/reducers';
+import { CastleFilterAction } from '../../../ngrx/actions';
+import { beginningOfYesterdayUTC, endOfYesterdayUTC, beginningOfLastWeekUTC, endOfLastWeekUTC,
+  beginningOfPriorTwoWeeksUTC, endOfPriorTwoWeeksUTC, beginningOfLastMonthUTC, endOfLastMonthUTC,
+  beginningOfPriorThreeMonthsUTC, endOfPriorThreeMonthsUTC, beginningOfLastYearUTC, endOfLastYearUTC,
+  getWhenForRange } from '../../../shared/util/date.util';
+
+@Component({
+  selector: 'metrics-prev-date-range',
+  template: `
+    <button class="btn-link" disabled="{{prevDisabled}}" (click)="prev()">&lt;&lt;&nbsp;PREV</button>
+  `
+})
+export class PrevDateRangeComponent implements OnInit, OnDestroy {
+  filterStoreSub: Subscription;
+  filter: FilterModel;
+
+  constructor(public store: Store<any>) {}
+
+  ngOnInit() {
+    this.filterStoreSub = this.store.select(selectFilter).subscribe(newFilter => {
+      this.filter = newFilter;
+    });
+  }
+
+  ngOnDestroy() {
+    if (this.filterStoreSub) {
+      this.filterStoreSub.unsubscribe();
+    }
+  }
+
+  get prevDisabled(): string {
+    if (!this.filter || !this.filter.range) {
+      return 'disabled';
+    } else {
+      return null;
+    }
+  }
+
+  prev() {
```

Comment:
was going to ask why the logic of getting the new date filters in this component doesn't look more like that of the next-date-range component -- i assume it's because navigating to previous time periods is a little trickier since we might not be in a complete one now? (e.g. if we're on "this week" but it's only 4 days into the week, couldn't navigate back by the same amount of days and expect it to say "last week") ?

Reply:
that's exactly it, it's for dealing with those partial ones for the standard ranges in the present

Reply:
👍